pkg/distributor/ha_tracker.go

```diff
@@ -141,50 +138,44 @@ func (r *ReplicaDesc) Clone() memberlist.Mergeable {
 // create an HA Tracker.
 type HATrackerConfig struct {
 	EnableHATracker bool `yaml:"enable_ha_tracker"`
-	// We should only update the timestamp if the difference
-	// between the stored timestamp and the time we received a sample at
-	// is more than this duration.
-	UpdateTimeout          time.Duration `yaml:"ha_tracker_update_timeout" category:"advanced"`
-	UpdateTimeoutJitterMax time.Duration `yaml:"ha_tracker_update_timeout_jitter_max" category:"advanced"`
-	// We should only failover to accepting samples from a replica
-	// other than the replica written in the KVStore if the difference
-	// between the stored timestamp and the time we received a sample is
-	// more than this duration
-	FailoverTimeout time.Duration `yaml:"ha_tracker_failover_timeout" category:"advanced"`
 
 	// This is a potentially high cardinality metric, so it is disabled by default.
 	EnableElectedReplicaMetric bool `yaml:"enable_elected_replica_metric"`
 
 	KVStore kv.Config `yaml:"kvstore" doc:"description=Backend storage to use for the ring. Note that memberlist support is experimental."`
+
+	DeprecatedHATrackerTimeoutsConfig `yaml:",inline"`
 }
 
 // RegisterFlags adds the flags required to config this to the given FlagSet.
 func (cfg *HATrackerConfig) RegisterFlags(f *flag.FlagSet) {
 	f.BoolVar(&cfg.EnableHATracker, "distributor.ha-tracker.enable", false, "Enable the distributors HA tracker so that it can accept samples from Prometheus HA replicas gracefully (requires labels).")
-	f.DurationVar(&cfg.UpdateTimeout, "distributor.ha-tracker.update-timeout", 15*time.Second, "Update the timestamp in the KV store for a given cluster/replica only after this amount of time has passed since the current stored timestamp.")
-	f.DurationVar(&cfg.UpdateTimeoutJitterMax, "distributor.ha-tracker.update-timeout-jitter-max", 5*time.Second, "Maximum jitter applied to the update timeout, in order to spread the HA heartbeats over time.")
-	f.DurationVar(&cfg.FailoverTimeout, "distributor.ha-tracker.failover-timeout", 30*time.Second, "If we don't receive any samples from the accepted replica for a cluster in this amount of time we will failover to the next replica we receive a sample from. This value must be greater than the update timeout")
 	f.BoolVar(&cfg.EnableElectedReplicaMetric, "distributor.ha-tracker.enable-elected-replica-metric", false, "Enable the elected_replica_status metric, which shows the current elected replica. It is disabled by default due to the possible high cardinality of the metric.")
 
+	// Even though those are deprecated in favor of Limits, we need to keep the flags
```

Comment:
@colega Could you let me know if this is clear enough? I realize it's hacky but I don't see how to make things fully backwards-compatible otherwise.

Reply:
No longer necessary after [2ab77ef](https://github.com/grafana/mimir/pull/11774/commits/2ab77eff772829566cb16468afbb0d43747561ca).